### PR TITLE
feat(string.hex): Hexadecimal validation to string

### PR DESCRIPTION
**Pull request by EmilEinarsen**
Details:

Added a method to the string schema for validating hexadecimal via regex

### CODE REVIEW

**Comment by dougiefresh49** [comment]
@EmilEinarsen found your pr from a google search, nice add! look forward to it. looks like you have a merge conflict in the `test/string.js`.

**Comment by EmilEinarsen** [comment]
@dougiefresh49 Thank you! I've resolved the merge conflict

**Comment by jaulz** [comment]
That would be cool!

File: src/string.ts [hunk 1]
```diff
@@ -25,8 +25,18 @@ let rUrl =
...
 
 // eslint-disable-next-line
+<<<<<<< HEAD
```
Comment (king-d-dev) [inline src/string.ts:28]: i guess you meant to get rid of this

File: src/string.ts [hunk 2]
```diff
@@ -25,8 +25,18 @@ let rUrl =
...
+// eslint-disable-next-line
+let rUUID = /^(?:[0-9a-f]{8}-[0-9a-f]{4}-[1-5][0-9a-f]{3}-[89ab][0-9a-f]{3}-[0-9a-f]{12}|00000000-0000-0000-0000-000000000000)$/i;
+// eslint-disable-next-line
+let rHex = /^#([0-9a-f]{3}|[0-9a-f]{6})$/i;
+=======
```
Comment (king-d-dev) [inline src/string.ts:36]: same here

File: src/string.ts [hunk 3]
```diff
@@ -25,8 +25,18 @@ let rUrl =
...
 let rUUID =
   /^(?:[0-9a-f]{8}-[0-9a-f]{4}-[1-5][0-9a-f]{3}-[89ab][0-9a-f]{3}-[0-9a-f]{12}|00000000-0000-0000-0000-000000000000)$/i;
+>>>>>>> master
```
Comment (king-d-dev) [inline src/string.ts:39]: same here

**Comment by slavafomin** [comment]
Hello! Thank you for the PR. @EmilEinarsen is there a chance you will fix it?

**Comment by jquense** [comment]
Not the most timely response sorry, thanks for the PR but generally we aren't adding new validation methods to the core API in order to keep it focused and lean. Yup allows for easy extensions tho so I encourage folks who find this useful to add in their own app or create a plugin

**Pull request closed without merging**